clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2096,7 +2099,7 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
 
   bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
     Expr *PointerRef =
-        createPointerParamReferenceExpr(FD->getType(), StructDepth != 0);
+        createPointerParamReferenceExpr(FieldTy, StructDepth != 0);
```

Comment:
This is because in case of arrays, FieldTy actually represents the value which we are visiting i.e. array element, not array field, right?

Reply:
Correct!